src/bokeh/models/annotations/legends.py

```diff
@@ -501,6 +514,142 @@ def __init__(self, *args, **kwargs) -> None:
     """).accepts(List(Tuple(String, List(Instance(GlyphRenderer)))),
         lambda items: [LegendItem(label=item[0], renderers=item[1]) for item in items])
 
+class ScaleBar(Annotation):
+    """ Represents a scale bar annotation.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    range = Either(Instance(Range), Auto, default="auto", help="""
+    The range for which to display the scale.
+
+    This can be either a range reference or ``"auto"``, in which case the
+    scale bar will pick the default x or y range of the frame, depending
+    on the orientation of the scale bar.
+    """)
+
+    unit = String(default="m", help="""
+    The unit of the ``range`` property.
+    """)
+
+    dimensional = Instance(Dimensional, default=InstanceDefault(MetricLength), help="""
+    Defines the units of measurement.
+    """)
+
+    orientation = Enum(Orientation, help="""
+    Whether the scale bar should be oriented horizontally or vertically.
+    """)
+
+    location = Enum(Anchor, default="top_right", help="""
+    Location anchor for positioning scale bar.
+    """)
+
+    length_sizing = Enum("adaptive", "exact", help="""
+    Defines how the length of the bar is interpreted.
+
+    This can either be:
+    * ``"adaptive"`` - the computed length is fit into a set of ticks provided
+        be the dimensional model. If no ticks are provided, then the behavior
+        is the same as if ``"exact"`` sizing was used
+    * ``"exact"`` - the computed length is used as-is
+    """)
+
+    bar_length = NonNegative(Either(Float, Int))(default=0.2, help="""
+    The length of the bar, either a percentage of the frame or a number of pixels.
+    """)
+
+    bar_style = Include(ScalarLineProps, prefix="bar", help="""
```

Comment:
The `_style` suffix seems unusual. Wouldn't `bar_props` be more consistent with existing terminology?

Reply:
Yes, though I changed this to `bar_line`, given there may be other visuals involved in the future.